controller/controller.go

```diff
@@ -186,7 +186,7 @@ type Controller struct {
 	// The interval between individual synchronizations
 	Interval time.Duration
 	// The DomainFilter defines which DNS records to keep or exclude
-	DomainFilter endpoint.DomainFilter
```

Comment:
Could someone explain why we changed the type here?

Reply:
At the time of implementing, there was a conflict with some providers.
It seems that this is not the case any longer and that this change was kept over rebases.
I tried reverting [b2ff1619](https://github.com/kubernetes-sigs/external-dns/pull/4597/commits/b2ff1619f5dfb50187fc75cb64733978d3a0d880) and it seems to work after some minor fixes.

What problems do you encounter because of this change?

Reply:
> What problems do you encounter because of this change?

It introduced a breaking change for some of the webhook providers as we used the DomainFilter type to simply Marshal the filter list for negotiation. We now just take the filter json and marshal it ourselves which is no big issue. It's just a loss of functionality. (writing this at 2am, might have a better way to explain it later)